DemoApp-Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ui/BaseAdActivity.java

```diff
@@ -0,0 +1,50 @@
+package com.applovin.enterprise.apps.demoapp.ui;
+
+import com.applovin.enterprise.apps.demoapp.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+/**
+ * Base for activities showing ads.
+ * <p>
+ * Created by Harry Arakkal on 2019-10-21.
+ */
+public abstract class BaseAdActivity
+        extends AppCompatActivity
+{
+    private CallbacksRecyclerViewAdapter callbacksAdapter;
+    private List<String>                 callbacks = new ArrayList<>();
+
+    /**
+     * Setup callbacks RecyclerView adapter and appearance.
+     */
+    protected void setupCallbacksRecyclerView()
```

Comment:
We have this super implement `protected void onCreate(Bundle savedInstanceState)` so we don't have to unnecessarily call `setupCallbacksRecyclerView()` in every subclass.

Reply:
I did it this way, because this function is dependent on `setContentView` being called in the subclasses first to find the `RecyclerView`. If we implemented this in `onCreate`, we would have to move the `super.onCreate` call after `setContentView` in all the subclasses. idk if that would have any side effects. If that's preferred, I can switch to using `onCreate`.